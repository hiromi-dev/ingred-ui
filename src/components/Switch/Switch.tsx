import * as React from "react";
import * as Styled from "./styled";
import Icon from "../Icon";
import { IconName } from "../Icon/Icon";
import Typography from "../Typography";
import Tooltip from "../Tooltip";

type Case = {
  name: string;
  value?: any;
};
type CaseWithIcon = Case & {
  icon: IconName;
};

type Props = {
  cases: Case[] | CaseWithIcon[];
  value: any;
  onChange?: (value: any) => void;
};

const Switch: React.FunctionComponent<Props> = ({ cases, value, onChange }) => {
  const items = cases as {
    name: string;
    icon?: IconName;
    value?: any;
  }[];
  const onHandleChange = (value: any) => () => {
    if (onChange) {
      onChange(value);
    }
  };
  return (
    <Styled.Container>
      {items.map((item, index) => {
        const childValue = item.value === undefined ? index : item.value;
        const active = childValue === value;
        return (
          <Tooltip
            key={item.name}
            content={item.name}
            positionPriority={["top-start"]}
            offset={[0, 15]}
          >
            {item.icon != null ? (
              <Styled.IconItemContainer
                active={active}
                onClick={onHandleChange(childValue)}
              >
                <Icon name={item.icon} size="lg" />
              </Styled.IconItemContainer>
            ) : (
              <Styled.TextItemContainer
                key={item.name}
                active={active}
                onClick={onHandleChange(childValue)}
              >
                <Typography
                  component="span"
                  size="sm"
                  weight={active ? "bold" : "normal"}
                  color="secondary"
                >
                  {item.name}
                </Typography>
              </Styled.TextItemContainer>
            )}
          </Tooltip>
        );
      })}
    </Styled.Container>
  );
};

export default Switch;
