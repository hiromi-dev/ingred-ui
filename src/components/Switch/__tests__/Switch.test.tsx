import * as React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";
import Switch from "../";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";

describe("Switch component testing", () => {
  afterEach(cleanup);

  test("Switch", () => {
    const { asFragment } = renderWithThemeProvider(
      <Switch
        cases={[
          {
            name: "デマンド別",
          },
          {
            name: "チャネル別",
          },
        ]}
        value={0}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
