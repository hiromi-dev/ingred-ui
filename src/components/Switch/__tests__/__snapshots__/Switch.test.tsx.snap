// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component testing Switch 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm gmOVYy"
  >
    <div
      class="sc-AxirZ gpWHsK"
    >
      <div
        class="sc-AxgMl jWlBRO"
        style="left: 0px; width: 0px;"
      />
      <div
        class="sc-AxiKw dELjYj"
      >
        <div
          class="sc-AxhUy eOzVgP"
        >
          <span
            class="sc-Axmtr eVWLQo"
            color="#596978"
            font-size="13px"
          >
            デマンド別
          </span>
        </div>
        <div
          class="sc-AxhUy eOzVgP"
        >
          <span
            class="sc-Axmtr kVmwSa"
            color="#596978"
            font-size="13px"
          >
            チャネル別
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
